src/webgpu/api/operation/rendering/draw.spec.ts

```diff
@@ -1,15 +1,27 @@
 export const description = `
 Tests for the general aspects of draw/drawIndexed/drawIndirect/drawIndexedIndirect.
 
-TODO: plan and implement
-- draws (note bind group state is not tested here):
-    - various zero-sized draws
-    - draws with vertexCount not aligned to primitive topology (line-list or triangle-list) (should not error)
-    - index buffer is {unset, set}
-    - vertex buffers are {unset, set} (some that the pipeline uses, some it doesn't)
-      (note: to test this, the shader in the pipeline doesn't have to actually use inputs)
-    - x= {draw, drawIndexed, drawIndirect, drawIndexedIndirect}
-- ?
+TODO:
+* basic - Test that basic draw functionality works
+     Note: compute a zero_sized private param. If true, check there are no shader invocations.
+       zero_sized is true if either the draw_size or the instance_count is zero.
```

Comment:
If it's computed, it can be computed in the function body instead of as a private param. Private params are only needed when you need to manually list them out alongside public params.